packages/pigeon/lib/generator_tools.dart

```diff
@@ -219,3 +219,29 @@ void addLines(Indent indent, Iterable<String> lines, {String? linePrefix}) {
     indent.writeln('$prefix$line');
   }
 }
+
+/// Recursively merges [y] into [x].  In other words, whenever there is a
+/// conflict over the value of a key path, [y] is selected.
+Map<String, Object> mergeMaps(Map<String, Object> x, Map<String, Object> y) {
```

Comment:
I didn't see any tests for this; it's a non-trivial utility method.

Reply:
added a test